pkg/services/ngalert/api/lotex_ruler.go

```diff
@@ -16,9 +16,32 @@ import (
 	"github.com/grafana/grafana/pkg/models"
 )
 
+const (
+	Prometheus = "prometheus"
+	Cortex     = "cortex"
+	Mimir      = "mimir"
+)
+
+const (
+	PrometheusDatasourceType = "prometheus"
+	LokiDatasourceType       = "loki"
+
+	mimirPrefix      = "/config/v1/rules"
+	prometheusPrefix = "/rules"
+	lokiPrefix       = "/api/prom/rules"
+
+	subtypeQuery = "subtype"
+)
+
 var dsTypeToRulerPrefix = map[string]string{
-	"prometheus": "/rules",
-	"loki":       "/api/prom/rules",
+	PrometheusDatasourceType: prometheusPrefix,
+	LokiDatasourceType:       lokiPrefix,
+}
+
+var subtypeToPrefix = map[string]string{
+	Prometheus: prometheusPrefix,
+	Cortex:     prometheusPrefix,
```

Comment:
Prometheus and Cortex share a path for the configuration API (and so does Mimir). However, there's more than meets the eye here (see more info in the original issue).

This complexity is meant to be tackled in #46771